### PR TITLE
improve password change and migrate logic

### LINKED ISSUE #428: Cannot decrypt some accounts after v6.0 update
Hi,
after the latest update v6.0 when I open the extension and I unlock it with password, not all my accounts are visible. Some of them remain encrypted.
Moreover, even if I setup a lock after some minutes, authenticator locks each time I close it.
I'm using the latest firefox v72.0.2 on GNU/linxu 64 bit.

### PR DESCRIPTION
**Pull request by mymindstorm**
fix #428

### CODE REVIEW

File: src/store/Accounts.ts [hunk 1]
```diff
@@ -320,16 +329,34 @@ export class Accounts implements IModule {
...
+                await entry.genUUID();
+              }
+            }
+
+            // store key
+            await BrowserStorage.set({
+              key: { enc: encKey, hash: encKeyHash }
+            });
+            await EntryStorage.set(state.state.entries);
+            if (removeHashes) {
```
Comment (Sneezry) [inline src/store/Accounts.ts]: This could be always true.

```js
console.log(!![]); // true
```

Comment (mymindstorm) [reply src/store/Accounts.ts]:
```suggestion
            if (removeHashes.length) {
```

File: src/store/Accounts.ts [hunk 2]
```diff
@@ -320,16 +329,34 @@ export class Accounts implements IModule {
...
             // change entry encryption and regen hash
+            let removeHashes: string[] = [];
             for (const entry of state.state.entries) {
               await entry.changeEncryption(
                 new Encryption(wordArray.toString())
               );
-              await entry.genUUID();
+              // if not uuidv4 regen
+              if (
+                !entry.hash.match(
```
Comment (Sneezry) [inline src/store/Accounts.ts]: Use `test` here makes more sense

File: src/store/Accounts.ts [hunk 3]
```diff
@@ -320,16 +329,34 @@ export class Accounts implements IModule {
...
+                  /[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}/i
+                )
+              ) {
+                removeHashes.push(entry.hash);
+                await entry.genUUID();
```
Comment (Sneezry) [inline src/store/Accounts.ts]: `genUUID` is a sync method

**Review by Sneezry — APPROVED** [review]

**Pull request merged**